spec/workers/attachments/extract_fulltext_job_spec.rb

```diff
@@ -102,28 +113,40 @@
   context "with the file not readable" do
     before do
       allow(attachment).to receive(:readable?).and_return(false)
+      perform_enqueued_jobs
     end
 
     include_examples "only file name indexed"
   end
 
-  context "with exception in extraction" do
+  context "with exception during extraction" do
     let(:exception_message) { "boom-internal-error" }
     let(:logger) { Rails.logger }
 
     before do
-      allow_any_instance_of(Plaintext::Resolver).to receive(:text).and_raise(exception_message)
+      allow_any_instance_of(Plaintext::Resolver).to receive(:text).and_raise(exception_message) # rubocop:disable RSpec/AnyInstance
 
       allow(logger).to receive(:error)
 
       allow(attachment).to receive(:readable?).and_return(true)
     end
 
-    it "logs the error" do
-      extracted_attributes
-      expect(logger).to have_received(:error).with(/boom-internal-error/)
+    it "raises the exception to update Job internal state and logs it" do
+      expect { perform_enqueued_jobs }.to raise_error(exception_message)
     end
 
-    include_examples "only file name indexed"
+    context "even with an exception being raised" do
```

Comment:
Looking at the `before`, is this a typo?

```suggestion
    context "even without an exception being raised" do
```

Reply:
No it's not a typo: the job is executed and raises an exception. It's rescued in the test because we don't want the test to end here.
And even if the job has raised an exception, a tsv has been added from the attachment filename, and the error was logged.

Reply:
I think what I find surprising about reading this context is that the raising of the exception is not related to the context. The parent context "with exception during extraction" is the one that introduces the raising of the exception, so it's not quite clear to me what behavioural change this context here is adding then. I guess none, except for "test implementation"-specifics.

My usual assumption about a context is, that it changes the test setup/premise, which it didn't here.

Reply:
yes, it's all about test implementation. I wanted to reuse the shared example, but it may be better without using it. I'll try rewording it.

Reply:
Let me throw this idea into the ring:

```suggestion
    context "when ignoring the raised exception" do
```

Reply:
Is it better now?

Reply:
Yes, much clearer IMO.